### ISSUE TITLE
Improve error handling - already used module id

### ISSUE THREAD

**Issue by breznak** [inilabs/caer#28]
As the `main.c` is rather large and gated by many #ifdefs I've added a function call which used an already in-use `moduleId` and got just a crash with core-dump. 

It would be nice if an error like this could be detected and reported properly, eg "The moduleId %i is already in use. Must be unique!"

**Comment by llongi** [comment]
Yes, I'm aware of that. Commit 675bc58fc9642efdbfe8471178b6326e44d67dfe fixes this, it just hasn't made it into master yet. I'll close this when I merge the branch back into master, which should be next week.

**Comment by llongi** [comment]
Merged into master.
